### PR TITLE
[py-pathy] new package

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the homepage and version sha256.

**Comment by tldahlgren** [comment]
Adding the python gurus/maintainers.

File: var/spack/repos/builtin/packages/py-pathy/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class PyPathy(PythonPackage):
+    """pathlib.Path subclasses for local and cloud bucket storage"""
+
+    homepage = "https://github.com/justindujardin/pathy"
+    pypi = "pathy/pathy-0.10.1.tar.gz"
+
+    version("0.10.1", sha256="4cd6e71b4cd5ff875cfbb949ad9fa5519d8d1dbe69d5fc1d1b23aa3cb049618b")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathy/package.py]: Spack doesn't support 3.5 and older, you can remove this line

File: var/spack/repos/builtin/packages/py-pathy/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on("py-setuptools", type="build")
+    depends_on("py-smart-open@5.2.1:6", type=("build", "run"))
+    depends_on("py-typer@0.3:0", type=("build", "run"))
+    depends_on("py-dataclasses@0.6:0", type=("build", "run"), when="^python@:3.6")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pathy/package.py]: Spack doesn't support Python 3.6 and older, you can remove this line

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**